### PR TITLE
lando: create initial django project with basic apps (bug 1864704)

### PR DESCRIPTION
**Pull request by zzzeid**
- add compose.yaml with db, proxy, and app services
- add Dockerfile for nginx and lando
- add nginx config
- add self-signed certificate generation + config
- update settings.py to support basic local setup
- add format command

### CODE REVIEW

File: nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+upstream lando {
```
Comment (cgsheeh) [inline nginx/nginx.conf:1]: This nginx config is just for running the prototype in Docker, right?

**Review by cgsheeh — CHANGES_REQUESTED** [review]
Why is there a `src/` folder that only contains `lando/`? In Python it's standard to just name the source repo after the project at the top level.

**Comment by zzzeid** [comment]
> Why is there a `src/` folder that only contains `lando/`? In Python it's standard to just name the source repo after the project at the top level.

This is a pretty common approach. See, for example: https://github.com/psf/requests, https://github.com/pallets/flask, https://github.com/mozilla-conduit/mots, and numerous others.

File: nginx/nginx.conf (hunk 1 above)
Comment (zzzeid) [reply nginx/nginx.conf:1]: Yes that is correct, it's used in `compose.yaml`.

**Comment by cgsheeh** [comment]
> > Why is there a `src/` folder that only contains `lando/`? In Python it's standard to just name the source repo after the project at the top level.
> 
> This is a pretty common approach. See, for example: https://github.com/psf/requests, https://github.com/pallets/flask, https://github.com/mozilla-conduit/mots, and numerous others.
> 
> NOTE: this is a preliminary project structure, in prep for more robust packaging which will be implemented later on.

It's a useless directory, regardless of who else has made one. :sweat_smile:

**Review by cgsheeh — APPROVED** [review]

**Comment by zzzeid** [comment]
> > > Why is there a `src/` folder that only contains `lando/`? In Python it's standard to just name the source repo after the project at the top level.
> > 
> > 
> > This is a pretty common approach. See, for example: https://github.com/psf/requests, https://github.com/pallets/flask, https://github.com/mozilla-conduit/mots, and numerous others.
> > NOTE: this is a preliminary project structure, in prep for more robust packaging which will be implemented later on.
> 
> It's a useless directory, regardless of who else has made one. 😅

https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/

**Comment by cgsheeh** [comment]
> > > > Why is there a `src/` folder that only contains `lando/`? In Python it's standard to just name the source repo after the project at the top level.
> > > 
> > > 
> > > This is a pretty common approach. See, for example: https://github.com/psf/requests, https://github.com/pallets/flask, https://github.com/mozilla-conduit/mots, and numerous others.
> > > NOTE: this is a preliminary project structure, in prep for more robust packaging which will be implemented later on.
> > 
> > 
> > It's a useless directory, regardless of who else has made one. 😅
> 
> https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/

Which of these differences is relevant to us?

**Comment by zzzeid** [comment]
> > > It's a useless directory, regardless of who else has made one. 😅
> > 
> > 
> > https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/
> 
> Which of these differences is relevant to us?

The link was in response to both "It's a useless directory" and "In Python it's standard to just name the source repo after the project at the top level." Both approaches are valid, and both have differences. Therefore the directory is not useless, and the other approach is not standard.

The main differences (mainly related to clear separation between "code being packaged" and "other code") won't be apparent until this project is a little more mature.

**Pull request merged**